test/remoteCollection/handleCollectResponse.js

```diff
@@ -136,50 +175,226 @@ describe('test/remoteCollection/handleCollectResponse.js >', () => {
     .catch(done);
   });
 
-  it('calls doBulkUpsert to push samples to refocus', (done) => {
-    // use nock to mock the response when flushing
-    const sampleArr = [
-      { name: 'S1|A1', value: 10 }, { name: 'S1|A2', value: 2 },
-    ];
-    nock(refocusUrl)
+  describe('handleCollectResponse', () => {
+
+    let winstonInfoStub;
+    before(() => {
+      // use nock to mock the response when flushing
+      const sampleArr = [
+        { name: 'S1|A1', value: 10 }, { name: 'S1|A2', value: 2 },
+      ];
+      nock(refocusUrl)
       .post(bulkEndPoint, sampleArr)
       .reply(httpStatus.CREATED, mockRest.bulkUpsertPostOk);
+
+      // stub winston info to test the logs
+      winstonInfoStub = sinon.stub(winston, 'info');
+    });
+
+    afterEach(() => {
+      winstonInfoStub.reset();
+      sampleQueueOps.flush(100, tu.refocusInstance1);
+    });
+
+    after(() => {
+      // restore winston stub
+      winstonInfoStub.restore();
+    });
+
     const collectRes = {
       name: 'mockGenerator',
       aspects: [{ name: 'A1', timeout: '1m' }, { name: 'A2', timeout: '1m' }],
       ctx: {},
-      res: { text: '{ "a": "atext" }' },
+      res: {
+        statusCode: 200,
+        statusMessage: 'MOCK STATUS MESSAGE',
+        body: {
+          text: '{ "a": "text" }',
+        },
+      },
       subjects: [{ absolutePath: 'S1' }],
```

Comment:
would be useful to test all of this with a subject with an absolutepath which contains a dot, to make sure the regex all work ... also, the test data for subjects should also have a "name" attribute

Reply:
good point, I'll change it.